app/controllers/images/originals_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Images::OriginalsController
+module Images
+  class OriginalsController < ApplicationController
+    # This is an AJAX request so we can keep overhead low.
+    disable_filters
+
+    def show
+      # Because this is only called from stimulus within valid pages, we
+      # can assume the user is already logged in and the image is correct...
+      # unless a malicious actor is calling this directly, in which case we
+      # don't really care if errors are handled gracefully.
+      @user = User.current = session_user or raise("login required")
+      @image = Image.safe_find(params[:id]) or raise("image not found")
+      log_request(@user, @image)
+
+      respond_to do |format|
+        format.html { redirect_to(@image.cached_original_url) }
+        format.json { cache_original_image }
+      end
+    end
+
+    def log_request(user, image)
+      OriginalImageRequest.create!(
+        user_id: user.id,
+        image_id: image.id
+      )
+    end
+
+    def cache_original_image
+      if on_image_server?
+        render(json: { status: "ready", url: @image.original_url })
+      elsif already_cached?
+        render(json: { status: "ready", url: @image.cached_original_url })
+      elsif already_loading?
+        render(json: { status: "loading" })
+      elsif user_maxed_out? || site_maxed_out?
+        render(json: { status: "maxed_out" })
+      else
+        # Save job for tester.
```

Comment:
I don't understand this comment.

Reply:
Would `# Save job for unit tests via assigns(:job).` make more sense?  Or is there a better way to check in the unit tests whether a job was queued?

Reply:
I don't really see the connection to the unit tests.  I must be missing something if `perform_later` is not intended to run in production.

Reply:
I want to test to make sure it actually queued a job in the unit test.  You're right, it's not intended to happen immediately, which is why it's perform_later.  (Actually, we wouldn't mind it happening immediately, but I assume one uses perform_later to ensure that it is done asynchronously, and that perform_now may literally block the execution thread?  But my intuition is frequently way off when it comes to rails!, so it's worth asking...)

But with perform_now it definitely doesn't happen immediately, so there's no way for the unit test to know by the *effects* of the job.  Therefore the unit test needs to be able to test for the existence of the (unperformed) job, right?  I don't care how.  This just seemed the easiest way that I personally knew how to do.